plugin/pkg/admission/serviceaccount/admission_test.go

```diff
@@ -877,348 +804,8 @@ func TestAddImagePullSecrets(t *testing.T) {
-		t.Fatal(err)
-	}
-
-	if n := len(pod.Spec.Volumes); n != 1 {
-		t.Fatalf("expected 1 volume mount, got %d", n)
-	}
-	if name := pod.Spec.Volumes[0].Name; name != token1 {
-		t.Errorf("expected first referenced secret to be mounted, got %q", name)
-	}
-}
-
-func newSecret(secretType corev1.SecretType, namespace, name, serviceAccountName, serviceAccountUID string) *corev1.Secret {
-	return &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: namespace,
-			Name:      name,
-			Annotations: map[string]string{
-				corev1.ServiceAccountNameKey: serviceAccountName,
-				corev1.ServiceAccountUIDKey:  serviceAccountUID,
-			},
-		},
-		Type: secretType,
-	}
-}
-
-func TestGetServiceAccountTokens(t *testing.T) {
-	testBoundServiceAccountTokenVolumePhases(t, func(t *testing.T, applyFeatures func(*Plugin) *Plugin) {
-		admit := applyFeatures(NewServiceAccount())
-		indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
-		admit.secretLister = corev1listers.NewSecretLister(indexer)
-
-		ns := "namespace"
-		serviceAccountUID := "12345"
-
-		sa := &corev1.ServiceAccount{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      DefaultServiceAccountName,
-				Namespace: ns,
-				UID:       types.UID(serviceAccountUID),
-			},
-		}
-
-		nonSATokenSecret := newSecret(corev1.SecretTypeDockercfg, ns, "nonSATokenSecret", DefaultServiceAccountName, serviceAccountUID)
-		indexer.Add(nonSATokenSecret)
-
-		differentSAToken := newSecret(corev1.SecretTypeServiceAccountToken, ns, "differentSAToken", "someOtherSA", "someOtherUID")
-		indexer.Add(differentSAToken)
-
-		matchingSAToken := newSecret(corev1.SecretTypeServiceAccountToken, ns, "matchingSAToken", DefaultServiceAccountName, serviceAccountUID)
-		indexer.Add(matchingSAToken)
-
-		tokens, err := admit.getServiceAccountTokens(sa)
-		if err != nil {
-			t.Fatalf("unexpected error: %v", err)
-		}
-
-		if len(tokens) != 1 {
-			names := make([]string, 0, len(tokens))
-			for _, token := range tokens {
-				names = append(names, token.Name)
-			}
-			t.Fatalf("expected only 1 token, got %v", names)
-		}
-		if e, a := matchingSAToken.Name, tokens[0].Name; e != a {
-			t.Errorf("expected token %s, got %s", e, a)
-		}
-	})
-}
-
-func TestAutomountIsBackwardsCompatible(t *testing.T) {
-	ns := "myns"
-	tokenName := "token-name"
-	serviceAccountName := DefaultServiceAccountName
-	serviceAccountUID := "12345"
-	defaultTokenName := "default-token-abc123"
-
-	expectedVolume := api.Volume{
-		Name: defaultTokenName,
-		VolumeSource: api.VolumeSource{
-			Secret: &api.SecretVolumeSource{
-				SecretName: defaultTokenName,
-			},
-		},
-	}
-	expectedVolumeMount := api.VolumeMount{
-		Name:      defaultTokenName,
-		ReadOnly:  true,
-		MountPath: DefaultAPITokenMountPath,
-	}
-
-	admit := NewServiceAccount()
-	admit.generateName = testGenerateName
-	admit.boundServiceAccountTokenVolume = deprecationEnabledBoundTokenVolume
```

Comment:
this is strange... was this testing admission with bound service account tokens turned on, and expecting a non-bound-service-account-token volume?

Reply:
oh... no... it's testing what happens if a pod that has a secret-based volume already passes through admission. I think we want to keep this test.

Reply:
initially i thought it was testing `getReferencedServiceAccountToken` then this test is not needed any more. 

as you mentioned, bound token was actually enabled in this test. looks like it just made sure existing volume at DefaultAPITokenMountPath would prevent bound token volume mounting, which is duplicate of `TestRespectsExistingMount`.